src/web/sitemap.rs

```diff
@@ -121,19 +164,72 @@ pub fn about_handler(req: &mut Request) -> IronResult<Response> {
 #[cfg(test)]
 mod tests {
     use crate::test::{assert_success, wrapper};
+    use reqwest::StatusCode;
 
     #[test]
-    fn sitemap() {
+    fn sitemap_index() {
         wrapper(|env| {
             let web = env.frontend();
-            assert_success("/sitemap.xml", web)?;
+            assert_success("/sitemap.xml", web)
+        })
+    }
+
+    #[test]
+    fn sitemap_invalid_letters() {
+        wrapper(|env| {
+            let web = env.frontend();
+
+            // everything not length=1 and ascii-lowercase should fail
+            for invalid_letter in &["1", "aa", "A", ""] {
+                println!("trying to fail letter {}", invalid_letter);
+                assert_eq!(
+                    web.get(&format!("/-/sitemap/{}/sitemap.xml", invalid_letter))
+                        .send()?
+                        .status(),
+                    StatusCode::NOT_FOUND
+                );
+            }
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn sitemap_letter() {
+        wrapper(|env| {
+            let web = env.frontend();
+
+            let letters: Vec<char> = (b'a'..=b'z').map(char::from).collect();
+
+            // letter-sitemaps always work, even without crates & releases
+            for letter in letters.iter().as_ref() {
```

Comment:
```suggestion
            // letter-sitemaps always work, even without crates & releases
            for letter in 'a'..='z' {
```

Reply:
@jyn514 I didn't know this works :) thanks!